### PR TITLE
 Use os structured fact instead of flat lsb facts

### PR DESCRIPTION
**Pull request by traylenator**
Scientific Linux CERN version has an
operatingsystem SLC on clone version 6.

### CODE REVIEW

**Comment by bastelfreak** [comment]
Thanks for the PR @traylenator, can you check the issues that travis found?

**Comment by brwyatt** [comment]
I don't think the build issues are related to the changes in this PR (or any of the other PRs). I looks (to me) that there's a few broken Ruby Gems getting pulled/used, or possibly issues with the metadata.json file.

Ruby versions before 2.x seem to be throwing a syntax error in the `metadata-json-lint` gem
Ruby versions 2.x are failing to find the `semantic_puppet` gem, then throw a warning about the 'pe' requirement in `metadata.json` no longer being supported by the forge.

So it looks like the fix will either be to remove support for Ruby <2.x, or use an older, more compatible, version of `metadata-json-lint`, add `semantic_puppet` to the `Gemfile` (I think), and remove/update the 'pe' requirement in `metadata.json`.

Either way, doesn't appear to be an issue with the changes in this PR, just looks like some Gems and `metadata.json` requirements were updated underneath the repo.

**Comment by bastelfreak** [comment]
@traylenator we recently did a modulesync. Can you please rebase?

**Comment by traylenator** [comment]
Happy to covert this to osfamily since so many changes going through anyway.

File: manifests/config.pp [hunk 1]
```diff
@@ -47,11 +47,10 @@
         require => $::fail2ban::config_file_require,
       }
     }
-    'Debian': {}
-    'Ubuntu': {
-      case $::lsbdistcodename {
+    'Debian': {
+      case $facts['os']['release']['major'] {
```
Comment (ekohl) [inline manifests/config.pp]: While you're at it, maybe change this to a simple if since there's just 1 case?

**Comment by traylenator** [comment]
And rebased.

**Pull request merged**